x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/pattern/index.tsx

```diff
@@ -272,10 +273,11 @@ const PatternComponent: React.FC<Props> = ({
         indices: getIndexNames({ stats, ilmExplain, ilmPhases, isILMAvailable }).length,
         pattern,
         results: undefined,
-        sizeInBytes: getTotalSizeInBytes({
-          indexNames: getIndexNames({ stats, ilmExplain, ilmPhases, isILMAvailable }),
-          stats,
-        }),
+        sizeInBytes:
+          getTotalSizeInBytes({
+            indexNames: getIndexNames({ stats, ilmExplain, ilmPhases, isILMAvailable }),
+            stats,
+          }) ?? getFallbackValue(isILMAvailable), // It fallback to 0 only when isILMAvailable is true
```

Comment:
Wouldn't be better to call `getTotalSizeInBytes` only when `isILMAvailable` is true? we already know we don't want to display sizes when it is false.

```suggestion
          isILMAvailable
          ? getTotalSizeInBytes({
              indexNames: getIndexNames({ stats, ilmExplain, ilmPhases, isILMAvailable }),
              stats,
            }) ?? 0
          : undefined,
```

the `getFallbackValue` won't be necessary.